[Gradient Compression] Add cuda.syncrhonize back to batched powerSGD

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54838 [Gradient Compression] Add cuda.syncrhonize back to batched powerSGD**

Realize that an explicit sync is somehow still needed for batched PowerSGD hook. I find that a job failure can be fixed by this change.

The sync was once removed by #54482.

Differential Revision: [D27384738](https://our.internmc.facebook.com/intern/diff/D27384738/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27384738/)!

Commit:
[Gradient Compression] Add cuda.syncrhonize back to batched powerSGD

Realize that an explicit sync is somehow still needed for batched PowerSGD hook. I find that a job failure can be fixed by this change.

The sync was once removed by #54482.

Differential Revision: [D27384738](https://our.internmc.facebook.com/intern/diff/D27384738/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27384738/)!

[ghstack-poisoned]

rohan-varma reviewed (approved):
This should be fine as a workaround to unblock experimentation for now. But ideally once this is shipped in a stable state, we would remove this synchronization as we generally try to avoid explicit synchronization for performance reasons, I believe. 

Is there an issue tracking a fix for this, or could you create one?

SciPioneer commented:
> This should be fine as a workaround to unblock experimentation for now. But ideally once this is shipped in a stable state, we would remove this synchronization as we generally try to avoid explicit synchronization for performance reasons, I believe.
> 
> Is there an issue tracking a fix for this, or could you create one?

I can see similar issue here:
https://github.com/pytorch/pytorch/issues/54842

I think for now explicit synchronization is somehow needed. I can link this PR to the code. Not sure if this can be fixed. Wdyt?

Commit:
Update on "[Gradient Compression] Add cuda.syncrhonize back to batched powerSGD"


Realize that an explicit sync is somehow still needed for batched PowerSGD hook. I find that a job failure can be fixed by this change.

The sync was once removed by #54482.

Differential Revision: [D27384738](https://our.internmc.facebook.com/intern/diff/D27384738/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27384738/)!

[ghstack-poisoned]
